mypyc/test-data/run-strings.test

```diff
@@ -178,3 +148,164 @@ def test_str_to_bool() -> None:
     for x in 'a', 'foo', 'bar', 'some string':
         assert is_true(x)
         assert not is_false(x)
+
+[case testFormattedStringLiterals]
+import decimal
+from datetime import datetime
+
+var = 'mypyc'
+num = 20
+
+def test_fstring_simple() -> None:
+    assert f'Hello {var}, this is a test' == "Hello mypyc, this is a test"
+
+def test_fstring_conversion() -> None:
+    assert f'Hello {var!r}' == "Hello 'mypyc'"
+    # repr() is equivalent to !r
+    assert f'Hello {repr(var)}' == "Hello 'mypyc'"
+
+    assert f'Hello {var!a}' == "Hello 'mypyc'"
+    # ascii() is equivalent to !a
+    assert f'Hello {ascii(var)}' == "Hello 'mypyc'"
+
+    tmp_str = """this
+    is a new line."""
+    assert f'Test: {tmp_str!a}' == "Test: 'this\\n    is a new line.'"
+
+    assert f'Hello {var!s}' == "Hello mypyc"
+    assert f'Hello {num!s}' == "Hello 20"
```

Comment:
Test an integer without `!s` (just `{}`). Test a negative integer. The big integer (>2**64), and also negative big integer.

Test `{}` with boolean, bytes and float arguments.